src/ffi.rs

```diff
@@ -217,7 +213,31 @@ pub unsafe extern "C" fn withdraw_stake(
 ) -> u8 {
     let refund = unwrap_or_bail!(PublicSpendKey::from_bytes(&*refund));
 
-    unwrap_or_bail!(WALLET.withdraw_stake(
+    unwrap_or_bail!(WALLET.unstake(
+        &mut FfiRng,
+        sender_index,
+        staker_index,
+        &refund,
+        gas_price,
+        gas_limit
+    ));
+
+    0
```

Comment:
I'm not sure what is strange about it. The `extern` functions all return an error code - 0 on success, anything else if not. If one of the functions wrapped by `unwrap_or_bail!` fails, for whatever reason, the appropriate error code is returned, otherwise the return could be bound by `let`.

Sometimes we're not interested in the return - which is the case here and perhaps why it looks strange - but sometimes we are. See for example here

https://github.com/dusk-network/wallet-core/blob/4ef79e608806a690f65d570d12b78e1fc42d7fd6/src/ffi.rs#L141-L150